.github/workflows/vast.yaml

```diff
@@ -1024,3 +1024,122 @@ jobs:
       - name: Run unit tests
         working-directory: python
         run: poetry run pytest
+
+  code-coverage:
+    needs:
+      - determine-version
+    if: github.event_name != 'workflow_dispatch'
+    name: Code Coverage
+    runs-on: ubuntu-20.04
+    container: debian:bullseye-slim
+    strategy:
+      fail-fast: false
+    env:
+      BUILD_DIR: build
+      CC: gcc-10
+      CXX: g++-10
+      CMAKE_GENERATOR: Ninja
+      CMAKE_MAKE_PROGRAM: ninja
+      DEBIAN_FRONTEND: noninteractive
+    steps:
+      - name: Install Dependencies
+        run: |
+          apt-get update
+          apt-get -y install \
+            apt-transport-https \
+            build-essential \
+            ca-certificates \
+            curl \
+            flatbuffers-compiler-dev \
+            g++-10 \
+            gcc-10 \
+            git \
+            gnupg2 gnupg-agent \
+            jq \
+            lcov \
+            libbroker-dev \
+            libcaf-dev \
+            libcaf-dev \
+            libflatbuffers-dev \
+            libfmt-dev \
+            libpcap-dev tcpdump \
+            libsimdjson-dev \
+            libspdlog-dev \
+            libssl-dev \
+            libunwind-dev \
+            libyaml-cpp-dev \
+            libxxhash-dev \
+            lsb-release \
+            ninja-build \
+            pandoc \
+            pkg-config \
+            python3-dev \
+            python3-pip \
+            python3-venv \
+            software-properties-common \
+            wget
+          # Apache Arrow (c.f. https://arrow.apache.org/install/)
+          wget "https://apache.jfrog.io/artifactory/arrow/$(lsb_release --id --short | tr 'A-Z' 'a-z')/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb" && \
+          apt-get -y install ./apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb && \
+          apt-get update
+          apt-get -y install libarrow-dev libparquet-dev
+          # Install CMake from pip -- we need at least 3.17 in CI for CCache
+          python3 -m pip install --upgrade pip
+          python3 -m pip install --upgrade cmake
+          cmake --version
```

Comment:
That's not possible because we don't upload that image in CI runs for PRs. I have some idea for reducing this down, but it requires a refactoring of our GitHub Actions workflow that I did not want to do in this PR: [Reusable Workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows).

Reply:
I was asking the same question as @lava. Glad you have already a follow-up planned, as this is hard to maintain.

Reply:
I'll write a follow-up story for deduping this somewhat, but it's hard to justify spending time on cleaning our workflow up.